rocky/tests/test_members.py

```diff
@@ -0,0 +1,162 @@
+        )
+
+
+def test_redteam_can_edit_itself(rf, redteam_member):
+    """
+    This will test if a redteam member can edit itself. Only admins and supersuers have edit rights.
+    """
+
+    request = setup_request(rf.get("organization_member_edit"), redteam_member.user)
+    with pytest.raises(PermissionDenied):
+        OrganizationMemberEditView.as_view()(
+            request, organization_code=redteam_member.organization.code, pk=redteam_member.id
+        )
+
+
+def test_admin_can_edit_superuser(rf, admin_member, superuser_member):
+    """
+    This will test if admin can edit superuser at the member edit view.
+    """
+
+    request = setup_request(rf.get("organization_member_edit"), admin_member.user)
+    with pytest.raises(PermissionDenied):
+        OrganizationMemberEditView.as_view()(
+            request, organization_code=superuser_member.organization.code, pk=superuser_member.id
+        )
+
+
+def test_client_can_edit_superuser(rf, client_member, superuser_member):
+    """
+    This will test if client can edit superuser at the member edit view.
+    """
+
+    request = setup_request(rf.get("organization_member_edit"), client_member.user)
+    with pytest.raises(PermissionDenied):
+        OrganizationMemberEditView.as_view()(
+            request, organization_code=superuser_member.organization.code, pk=superuser_member.id
+        )
+
+
+def test_redteamer_can_edit_superuser(rf, redteam_member, superuser_member, organization):
+    """
+    This will test if redteamer can edit superuser at the member edit view.
+    """
+
+    request = setup_request(rf.get("organization_member_edit"), redteam_member.user)
+    with pytest.raises(PermissionDenied):
+        OrganizationMemberEditView.as_view()(request, organization_code=organization.code, pk=superuser_member.id)
+
+
+def test_edit_superusers_from_different_organizations(rf, superuser_member, superuser_member_b):
+    """
+    This will test if a superuser from one organization can edit
+    a superuser from another organization at the member edit view.
+    """
+
+    request = setup_request(rf.get("organization_member_edit"), superuser_member.user)
+    # from OrganizationView
+    with pytest.raises(Http404):
+        OrganizationMemberEditView.as_view()(
+            request, organization_code=superuser_member_b.organization.code, pk=superuser_member_b.id
+        )
+
+
+def test_edit_admins_from_different_organizations(rf, admin_member, admin_member_b):
+    """
+    This will test if a admin from one organization can edit
+    a admin from another organization at the member edit view.
+    """
+
+    request = setup_request(rf.get("organization_member_edit"), admin_member.user)
+    # from OrganizationView
+    with pytest.raises(Http404):
+        OrganizationMemberEditView.as_view()(
+            request, organization_code=admin_member_b.organization.code, pk=admin_member_b.id
+        )
```

Comment:
What is missing is the case where the user is an admin for organization A but a client for organization B and tries to edit B.

What is also missing is test for the actual post that changes things.

Reply:
See commit: https://github.com/minvws/nl-kat-coordination/pull/318/commits/d2b2b4d21514943d91be4767237549db64ea86e6

Reply:
What you test there is user that is an admin of organization A and not a member of organization B. That is different from a user that is an admin of organization A and a non-admin member or organiation B.

Reply:
You are indeed right. Here admin member of orgA is requesting the MemberEditView of another admin member from orgB using the pk=admin_member_b.id  which will throw this 404, either way if the user is a client or a redteamer it will have the same effect. This comes form the OrganizationView that a user/member cannot even view things from another org. We can remove this test or do you have any idea how to mock this?

Reply:
The issue is when the user is an admin member of orgA and **also a non-admin member of orgB**. This will not throw a 404, because the user is a member of orgB. Permissions are global, so this user will also have the "tools.change_organizationmember" permission. So this user will be able to edit organization members while not being an admin member of orgB.